internal/codegen/template.go

```diff
@@ -126,9 +126,30 @@ func (t *Template) Render(st *Stencil, vals *Values, dms map[string]*DirManifest
 		t.Files = t.Files[1:len(t.Files)]
 	}
 
-	// Now that everything's been decided, see if we need to replace any file paths from directory manifests
-	for _, tf := range t.Files {
-		tf.ApplyDirManifests(dms)
+	if !st.isFirstPass {
+		// Now that everything's been decided, see if we need to replace any file paths from directory manifests
+		for _, tf := range t.Files {
+			// Hop through the path dir by dir, starting at the end (because the raw paths won't match if you replace the earlier
+			// path segments first), and see if there's any replacements.
+			pp := strings.Split(tf.path, string(os.PathSeparator))
+			for i := len(pp) - 1; i >= 0; i-- {
+				pathPart := strings.Join(pp[0:i+1], string(os.PathSeparator))
+				if dr, has := t.Module.Manifest.DirReplacements[pathPart]; has {
```

Comment:
Can you break this out into it's own function? I don't like the level of nesting here (also nested for-loops are a must avoid, IMO.

Reply:
Might be worth breaking out the entire thing also, to be honest.